config/watcher.go

```diff
@@ -142,11 +142,20 @@ func newFSWatcher(ctx context.Context, configPath string, logger logging.Logger,
 			case <-cancelCtx.Done():
 				return
 			case event := <-fsWatcher.Events:
-				if event.Op&fsnotify.Write == fsnotify.Write {
+				// Monitor WRITE and REMOVE events.
+				// Editors that save in place WRITE over the monitored file.
+				// Editors that save atomically write to a temp file and swap. Events on original file are: RENAME->CHMOD->REMOVE
+				if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Remove == fsnotify.Remove {
 					debounced(func() {
 						logger.Info("On-disk config file changed. Reloading the config file.")
 						//nolint:gosec
 						rd, err := os.ReadFile(configPath)
+
+						// Re-add to watcher. Will be a new inode if it was saved atomically.
```

Comment:
Just out of my own curiosity, will the path of the new file not be identical to the path of the old file? I buy that the "atomic save" results in a differing entry in the inode table for the system, but why would re-adding the same path to the watcher account for that?

Reply:
Awesome; I think that makes sense. Thanks.